lib/src/vector_math/matrix4.dart

```diff
@@ -1992,7 +1992,9 @@ class Matrix4 {
 
   /// Multiply this to each set of xyz values in [array] starting at [offset].
   List<double> applyToVector3Array(List<double> array, [int offset = 0]) {
-    for (int i = 0, j = offset; i < array.length; i += 3, j += 3) {
+    for (int i = 0, j = offset;
+        i < array.length && j < array.length;
+        i += 3, j += 3) {
```

Comment:
`i` is not used in the loop - remove it (and then maybe rename `j` to `i`).

The API assumes that `array` is length 3N and that `offset` is a multiple of 3.
These constraints should be added to the documentation.

It is possible to make the function safe for improper lengths or offsets:
```
for (int j = offset; j + 2 < array.length; j += 3) {
  ...
}
```